### PR TITLE
feat(konflux): add prerelease GPG check disable for RPM prefetch

### PR DESCRIPTION
**Pull request by lgarciaaco**
## Summary
Extended RPM prefetch functionality to conditionally disable GPG checking during prerelease phase by adding DNF repository configuration with gpgcheck=0 for all enabled repositories using proper repository IDs.

## Problem
**Before:** Current RPM prefetch only includes basic structure `{"type": "rpm", "path": "."}` without any DNF repository configuration options, regardless of software lifecycle phase.
**After:** When building images during prerelease phase, RPM prefetch includes DNF repository configuration that disables GPG checking (`"gpgcheck": "0"`) for all repositories, following the Hermeto specification format with proper content set IDs.

## Changes
- `doozer/doozerlib/backend/konflux_image_builder.py`: Enhanced _prefetch() method (lines 393-416) to detect prerelease phase and add DNF repository configuration with proper repository ID extraction
- `doozer/tests/backend/test_konflux_cachi2.py`: Added 5 comprehensive test cases covering prerelease functionality, edge cases, and repository ID validation

**Technical Notes**
- Uses `metadata.runtime.group_config.software_lifecycle.phase == 'pre-release'` for phase detection  
- Accesses repository objects via public `metadata.runtime.repos[repo_name]` interface
- Extracts proper content set IDs using `repo.content_set(arch)` method for all building architectures
- Includes fallback logic when content_set returns None (same pattern as lockfile generation)
- Generates prefetch structure: `{"type": "rpm", "path": ".", "options": {"dnf": {"repo-id": {"gpgcheck": "0"}}}}`
- Maintains backward compatibility - zero impact on non-prerelease builds

**Example Output:**
```json
{
  "type": "rpm",
  "path": ".",
  "options": {
    "dnf": {
      "fast-datapath-for-rhel-9-aarch64-rpms": {"gpgcheck": "0"},
      "fast-datapath-for-rhel-9-x86_64-rpms": {"gpgcheck": "0"},
      "rhel-9-for-aarch64-appstream-eus-rpms__9_DOT_4": {"gpgcheck": "0"},
      "rhel-9-for-aarch64-baseos-eus-rpms__9_DOT_4": {"gpgcheck": "0"},
      "rhel-9-for-x86_64-appstream-eus-rpms__9_DOT_4": {"gpgcheck": "0"},
      "rhel-9-for-x86_64-baseos-eus-rpms__9_DOT_4": {"gpgcheck": "0"},
      "rhocp-4.21-for-rhel-9-aarch64-rpms": {"gpgcheck": "0"},
      "rhocp-4.21-for-rhel-9-x86_64-rpms": {"gpgcheck": "0"}
    }
  }
}
```

**Related:**
- Successfully tested with hermetic build: https://konflux-ui.apps.kflux-ocp-p01.7ayg.p1.openshiftapps.com/ns/ocp-art-tenant/applications/openshift-4-21/pipelineruns/ose-4-21-dpu-intel-ipu-vsp-26m8b/
- Closes: https://issues.redhat.com/browse/ART-14028

### CODE REVIEW

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 1]
```diff
@@ -389,6 +389,36 @@ def _prefetch(self, metadata: ImageMetadata, dest_dir: Optional[Path] = None) ->
                 "type": "rpm",
                 "path": lockfile_path,
             }
+
+            # Add prerelease-specific DNF configuration
+            phase = metadata.runtime.group_config.software_lifecycle.phase
```
[the remark this answers is not in the document]
Comment (lgarciaaco) [reply doozer/doozerlib/backend/konflux_image_builder.py:394]: @vfreex I do not want to replace a single access to a dict key(current implementation), with a call to an async function which itself call `doozer` command down the line to achieve the same, as your suggestion is.

If there is a better way to achieve this I gladly do the change, or if you think my implementation is faulty tell me what is missing

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 2]
```diff
@@ -389,6 +389,36 @@ def _prefetch(self, metadata: ImageMetadata, dest_dir: Optional[Path] = None) ->
...
+                    repos = metadata.runtime.repos  # Repos instance
+                    building_arches = metadata.get_arches()
+
+                    for repo_name in enabled_repos:
+                        repo = repos[repo_name]  # Use public interface
+                        for arch in building_arches:
+                            # Only apply to repositories with baseurl containing '/plashets/'
+                            baseurl = repo.baseurl("unsigned", arch)
+                            if "/plashets/" not in baseurl:
+                                continue
```
[the remark this answers is not in the document]
Comment (lgarciaaco) [reply doozer/doozerlib/backend/konflux_image_builder.py]: Addressed with https://github.com/openshift-eng/art-tools/pull/2030/commits/3aef43605acf64a2b6f485960ea91f7e4f6cfc2c

File: doozer/doozerlib/backend/konflux_image_builder.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vfreex) [reply doozer/doozerlib/backend/konflux_image_builder.py:393]: I see. Maybe copying some lines from `get_signing_mode`:

```python
phase = SoftwareLifecyclePhase.from_name(group_config['software_lifecycle']['phase'])
gpgcheck = if phase >= SoftwareLifecyclePhase.SIGNING
```
